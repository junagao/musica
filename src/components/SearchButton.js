import React from 'react';
import { IoIosSearch } from 'react-icons/io';

import './SearchButton.scss';

const SearchButton = () => (
  <button type="submit" className="search-button">
    <IoIosSearch />
  </button>
);

export default SearchButton;
