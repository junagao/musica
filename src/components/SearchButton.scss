@import '../variables';

.search-button {
  background-color: $color-main;
  border-radius: 0.5rem;
  border: none;
  color: $color-icon;
  cursor: pointer;
  font-size: 2.5rem;
  outline: none;
  padding: 0.4rem 0.3rem 0rem 0.3rem;
  width: 54px;
  height: 54px;
  &:hover {
    background-color: $color-darker;
  }
}
