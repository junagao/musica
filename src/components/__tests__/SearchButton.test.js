import React from 'react';
import { shallow } from 'enzyme';
import { IoIosSearch } from 'react-icons/io';

import SearchButton from 'components/SearchButton';

describe('SearchButton', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<SearchButton />);
  });

  it('should render correctly given the required props', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render one button element of type `submit` and className `search-button`', () => {
    expect(wrapper.find('button').length).toEqual(1);
    expect(wrapper.find('button').prop('type')).toEqual('submit');
    expect(wrapper.find('button').hasClass('search-button')).toBeTruthy();
  });

  it('should render IoIosSearch icon', () => {
    expect(wrapper.find('button').text()).toEqual('<IoIosSearch />');
    expect(
      wrapper.containsMatchingElement(
        <button type="submit" className="search-button">
          <IoIosSearch />
        </button>,
      ),
    ).toBeTruthy();
  });
});
