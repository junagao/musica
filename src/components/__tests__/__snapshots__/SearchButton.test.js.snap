// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchButton should render correctly given the required props 1`] = `
<button
  className="search-button"
  type="submit"
>
  <IoIosSearch />
</button>
`;
